### PR TITLE
backport: 19754 and necessary pre-reqs

### PR DESCRIPTION
**Pull request by PastaPastaPasta**

### CODE REVIEW

**Review by kittywhiskers — CHANGES_REQUESTED** [review]
See https://github.com/kittywhiskers/dash/commits/dash4880, rebased over develop, squashed Udjin's fixes and resolved `test_dash_qt` failure. Also resolved incorrect address bug caused by querying `GetDataDir()` instead of `GetWalletDir()`.

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -2814,6 +2831,8 @@ static UniValue createwallet(const JSONRPCRequest& request)
         // no default case, so the compiler can warn about missing cases
     }
 
+    UpdateWalletSetting(*context.chain, request.params[0].get_str(), request.params[6], warnings);
```
[the remark this answers is not in the document]
Comment (PastaPastaPasta) [reply src/wallet/rpcwallet.cpp]: Looks like this got removed in a later PR?

File: src/wallet/rpcwallet.cpp [hunk 2]
```diff
@@ -2820,7 +2807,8 @@ static UniValue createwallet(const JSONRPCRequest& request)
 
     bilingual_str error;
     std::shared_ptr<CWallet> wallet;
-    WalletCreationStatus status = CreateWallet(*context.chain, passphrase, flags, request.params[0].get_str(), error, warnings, wallet);
+    Optional<bool> load_on_start = request.params[4].isNull() ? nullopt : Optional<bool>(request.params[4].get_bool());
```
[the remark this answers is not in the document]
Comment (PastaPastaPasta) [reply src/wallet/rpcwallet.cpp]: applied and squashed

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/wallet/rpcwallet.cpp [hunk 3]
```diff
@@ -2820,7 +2807,8 @@ static UniValue createwallet(const JSONRPCRequest& request)
 
     bilingual_str error;
     std::shared_ptr<CWallet> wallet;
-    WalletCreationStatus status = CreateWallet(*context.chain, passphrase, flags, request.params[0].get_str(), error, warnings, wallet);
+    Optional<bool> load_on_start = request.params[5].isNull() ? nullopt : Optional<bool>(request.params[4].get_bool());
```
[the remark this answers is not in the document]
Comment (PastaPastaPasta) [reply src/wallet/rpcwallet.cpp:2810]: whoops

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**